File: 200_LeetCode_Questions/Java/D04P2_LeetCode_1395.java
// Runtime: 3 ms, faster than 75.45% of Java online submissions for Count Number of Teams.
// Memory Usage: 37.1 MB, less than 67.41% of Java online submissions for Count Number of Teams.

class Solution {
    public int numTeams(int[] rating) {
        int teams = 0;
        int[] low;
        int[] high;
        for (int i = 1; i < rating.length - 1; ++i) {
            low = new int[2];
            high = new int[2];
            for (int j = 0; j < rating.length; ++j) {
                if (rating[i] < rating[j]){
                    ++low[j > i ? 1 : 0];
                }
                if (rating[i] > rating[j]){
                    ++high[j > i ? 1 : 0];
                }
            }
            teams += low[0] * high[1] + high[0] * low[1];
        }
        return teams;
    }
}

File: 200_LeetCode_Questions/Python/D04P2_LeetCode_1395.py
class Solution:
    def numTeams(self, rating: List[int]) -> int:
        count=0
        for  i in range (len(rating)):
            for j in range(i+1,len(rating)):
                for k in range(j+1,len(rating)):
                    if (rating[i] < rating[j] < rating[k]) or (rating[i] > rating[j] > rating[k]):
                        count+=1
        return count
                    
                    
